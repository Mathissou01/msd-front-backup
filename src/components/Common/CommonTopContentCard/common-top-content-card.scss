@use "src/styles/01-settings/settings.colors" as colors;
@use "src/styles/02-tools/tools.typography" as typo;
@use "src/styles/02-tools/tools.inputs" as input;
@use "src/styles/02-tools/tools.media-queries" as media;
@use "src/styles/02-tools/tools.shadows" as shadows;
@use "src/styles/02-tools/tools.z-index" as index;

.c-CommonTopContentCard {
  position: relative;
  z-index: index.z("site-content");

  @include media.from-tablet {
    position: relative;
    width: 784px;
    height: 380px;
    padding: 36px 0;
  }

  &__Image {
    position: relative;
    display: flex;
    align-items: center;
    justify-content: center;

    @include media.mobile {
      min-width: 375px;
      height: 205px;

      img {
        width: 100%;
        height: 100%;
      }
    }

    @include media.small-tablet {
      width: 480px;
      max-height: 304px;

      img {
        width: 100%;
        max-width: 480px;
        height: 100%;
      }
    }

    @include media.from-tablet {
      max-width: 480px;
      height: 308px;
      border-radius: 8px;

      img {
        width: 480px;
        max-width: 480px;
        height: 308px;
      }
    }

    &_hoverImg {
      transition: transform 0.2s ease-in-out;

      &:hover {
        transform: scale(1.1);
      }
    }
  }

  &__Img {
    &_mobile {
      display: initial;

      @include media.from-tablet {
        display: none;
      }
    }

    &_desktop {
      display: none;

      @include media.from-tablet {
        display: initial;
        border-radius: 8px;
      }
    }
  }

  &__ContentContainer {
    @include media.from-tablet {
      position: absolute;
      top: 50px;
      right: 18px;
      display: flex;
      align-items: center;
      height: 308px;
    }
  }

  &__Content {
    @include shadows.shadow(1);

    position: relative;
    bottom: 17px;
    left: 18px;
    display: flex;
    flex-direction: column;
    gap: 16px;
    padding: 24px 16px;
    background-color: colors.$white;
    border-radius: 8px;

    @include media.mobile {
      max-width: 337px;
    }

    @include media.small-tablet {
      left: 25px;
      max-width: 431px;
    }

    @include media.from-tablet {
      width: 413px;
      height: fit-content;
      padding: 24px 32px;
    }
  }

  &__ContentTags {
    display: flex;
    align-items: center;
    justify-content: space-between;
  }

  &__Tags {
    display: flex;
    gap: 6px;
    padding: 0;
  }

  &__Tag {
    @include typo.text("tag");

    padding: 6px 10px;
    color: colors.$expert-blue;
    border: 1px solid colors.$blue-4;
    border-radius: 4px;

    &_background {
      background-color: colors.$gray-0;
      border: hidden;
    }
  }

  &__ContentCardDate {
    @include typo.text("tag");

    text-transform: lowercase;
  }

  &__Description {
    display: flex;
    flex-direction: column;
    gap: 6px;
    padding: 0;
  }

  &__TitleContent {
    @include typo.heading(3);
  }

  &__TitleContentTop {
    @include typo.heading(1);
  }

  &__DescriptionContent {
    color: colors.$gray-800;
    font-weight: 400;
    font-size: 14px;
    line-height: 21px;
  }

  &__Link {
    display: flex;
    gap: 12px;
    align-items: center;
    padding: 0;
    border-radius: 2px;
    transition: text-decoration 0.3s ease-in-out;

    &:hover {
      text-decoration: underline;
    }
  }

  &__Label {
    @include typo.text("caption");

    color: colors.$blue-1;

    @include media.from-tablet {
      @include typo.text("buttonLarge");
    }
  }

  &__EditoContent {
    @include media.from-tablet {
      width: 100%;
    }

    .c-CommonTopContentCard__ContentContainer {
      width: 100%;

      @include media.from-tablet {
        width: 74%;
      }
    }

    .c-CommonTopContentCard__Content {
      width: 100%;

      img {
        height: 297px;
      }
    }

    svg {
      #primary-color {
        fill: colors.$brand-primary;
      }

      #secondary-color {
        fill: colors.$brand-secondary;
      }
    }
  }
}
