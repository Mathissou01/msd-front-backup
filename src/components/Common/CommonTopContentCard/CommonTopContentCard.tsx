import classNames from "classnames";
import React from "react";
import Link from "next/link";
import Image from "next/image";
import { TagEntity } from "../../../graphql/codegen/generated-types";
import { handleDateFrenchFormat } from "../../../lib/utilities";
import ArrowCorner from "public/images/pictos/arrow-corner.svg";
import "./common-top-content-card.scss";

interface ICommonTopContentCardProps {
  title: string;
  redirectUrl: string;
  description?: string;
  tags?: TagEntity[];
  date?: string;
  imageUrlDesktop?: string;
  imageUrlMobile?: string;
  imageAlt?: string;
  isTitleTop?: boolean;
  style?: "default" | "editoPage";
  isImgHasHover?: boolean;
}

export default function CommonTopContentCard({
  title,
  redirectUrl,
  description,
  tags,
  date,
  imageUrlDesktop,
  imageUrlMobile,
  imageAlt = "",
  isTitleTop = false,
  isImgHasHover = false,
  style = "default",
}: ICommonTopContentCardProps) {
  const knowMore = "En savoir plus";

  let contentCardDate;
  let dataFrenchFormat;
  if (date) {
    contentCardDate = new Date(date || "");
    dataFrenchFormat = handleDateFrenchFormat(contentCardDate);
  }

  const cardClassNames = classNames("c-CommonTopContentCard", {
    "c-CommonTopContentCard__EditoContent": style === "editoPage",
  });

  const imageClassNames = classNames("c-CommonTopContentCard__Image", {
    "c-CommonTopContentCard__Image_hoverImg": isImgHasHover,
  });

  const getTagClassNames = (index: number) => {
    return classNames("c-CommonTopContentCard__Tag", {
      "c-CommonTopContentCard__Tag_background": index > 0,
    });
  };

  return (
    <div className={cardClassNames}>
      <div className={imageClassNames}>
        {imageUrlDesktop ? (
          <Image
            src={imageUrlDesktop}
            alt={imageAlt}
            width={480}
            height={308}
            className="c-CommonTopContentCard__Img_desktop"
          />
        ) : null}
        {imageUrlMobile ? (
          <Image
            src={imageUrlMobile}
            alt={imageAlt}
            width={312}
            height={200}
            className="c-CommonTopContentCard__Img_mobile"
          />
        ) : null}
      </div>
      <div className="c-CommonTopContentCard__ContentContainer">
        <div className="c-CommonTopContentCard__Content">
          {isTitleTop ? (
            <h2 className="c-CommonTopContentCard__TitleContentTop">{title}</h2>
          ) : null}
          <div className="c-CommonTopContentCard__ContentTags">
            <div className="c-CommonTopContentCard__Tags">
              {tags?.map((tag, index) => (
                <span
                  key={tag.attributes?.name}
                  className={getTagClassNames(index)}
                >
                  {tag.attributes?.name}
                </span>
              ))}
            </div>
            {!isTitleTop ? (
              <div className="c-CommonTopContentCard__ContentCardDate">
                {dataFrenchFormat}
              </div>
            ) : null}
          </div>
          <div className="c-CommonTopContentCard__Description">
            {!isTitleTop ? (
              <h2 className="c-CommonTopContentCard__TitleContent">{title}</h2>
            ) : null}
            <p className="c-CommonTopContentCard__DescriptionContent">
              {description}
            </p>
          </div>
          {!isTitleTop ? (
            <Link className="c-CommonTopContentCard__Link" href={redirectUrl}>
              <span className="c-CommonTopContentCard__Label">{knowMore}</span>
              <ArrowCorner />
            </Link>
          ) : null}
        </div>
      </div>
    </div>
  );
}
