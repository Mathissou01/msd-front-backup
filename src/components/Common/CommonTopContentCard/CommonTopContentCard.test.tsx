import { render } from "@testing-library/react";
import CommonTopContentCard from "./CommonTopContentCard";

const mock = {
  title: "common top content card title",
  href: "/accueil",
  imageUrl: "/images/pictos/default.svg",
};

describe("CommonTopContentCard", () => {
  it("renders", () => {
    const { container } = render(
      <CommonTopContentCard
        title={mock.title}
        redirectUrl={mock.href}
        imageUrlDesktop={mock.imageUrl}
      />,
    );
    expect(container).toMatchSnapshot();
  });
});
