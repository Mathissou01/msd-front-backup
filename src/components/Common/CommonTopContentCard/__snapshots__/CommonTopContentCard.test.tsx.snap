// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommonTopContentCard renders 1`] = `
<div>
  <div
    class="c-CommonTopContentCard"
  >
    <div
      class="c-CommonTopContentCard__Image"
    >
      <img
        alt=""
        class="c-CommonTopContentCard__Img_desktop"
        data-nimg="1"
        decoding="async"
        height="308"
        loading="lazy"
        src="/images/pictos/default.svg"
        style="color: transparent;"
        width="480"
      />
    </div>
    <div
      class="c-CommonTopContentCard__ContentContainer"
    >
      <div
        class="c-CommonTopContentCard__Content"
      >
        <div
          class="c-CommonTopContentCard__ContentTags"
        >
          <div
            class="c-CommonTopContentCard__Tags"
          />
          <div
            class="c-CommonTopContentCard__ContentCardDate"
          />
        </div>
        <div
          class="c-CommonTopContentCard__Description"
        >
          <h2
            class="c-CommonTopContentCard__TitleContent"
          >
            common top content card title
          </h2>
          <p
            class="c-CommonTopContentCard__DescriptionContent"
          />
        </div>
        <a
          class="c-CommonTopContentCard__Link"
          href="/accueil"
        >
          <span
            class="c-CommonTopContentCard__Label"
          >
            En savoir plus
          </span>
          <div />
        </a>
      </div>
    </div>
  </div>
</div>
`;
